src/test/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceTest.java

```diff
@@ -208,6 +209,34 @@ void testGetByReferenceIdForbidden() {
     assertThrows(ForbiddenException.class, () -> resource.getByReferenceId(authUser, ""));
   }
 
+  @ParameterizedTest
+  @EnumSource(value = UserRoles.class, names = {"ADMIN", "CHAIRPERSON", "MEMBER", "SIGNINGOFFICIAL"})
+  void testGetByReferenceIdAllowedRoles(UserRoles role) {
+    UserRole userRole = new UserRole(role.getRoleId(), role.getRoleName());
+    User roleUser = new User(1, authUser.getEmail(), "Display Name", new Date(), List.of(userRole));
+    when(userService.findUserByEmail(roleUser.getEmail())).thenReturn(roleUser);
+    // Set the DAR create user to be a different user from the roleUser
+    DataAccessRequest dar = generateDataAccessRequest();
+    dar.setUserId(roleUser.getUserId() + 1);
+    when(dataAccessRequestService.findByReferenceId(any())).thenReturn(dar);
+
+    Response response = resource.getByReferenceId(authUser, "");
```

Comment:
You could replace `any` with an ID here
```suggestion
    when(dataAccessRequestService.findByReferenceId("id")).thenReturn(dar);

    Response response = resource.getByReferenceId(authUser, "id");
```